src/index.tsx

```diff
@@ -508,22 +510,26 @@ export function useScreencast() {
  * a function that updates endpoints's metadata on the server
  * @param metadata a map `string -> any` containing user's track metadata to be sent to the server
  */
-export async function updateEndpointMetadata(metadata: Metadata) {
+export async function updateEndpointMetadata<T extends Metadata>(metadata: T) {
   await MembraneWebRTCModule.updateEndpointMetadata(metadata);
 }
 
 /**
  * a function that updates video metadata on the server.
  * @param metadata a map string -> any containing video track metadata to be sent to the server
  */
-export async function updateVideoTrackMetadata(metadata: Metadata) {
+export async function updateVideoTrackMetadata<T extends Metadata>(
```

Comment:
there is a problem with that: I can specify different type for metadata when starting the camera and when updating camera track metadata
ideally it'd be inferred automatically but I don't know how to do this or if it's possible (maybe with some api changes?)